File: python/bob/bob.py
import re


def response(hey_bob: str) -> str:
    question = hey_bob.rstrip().endswith("?")
    yelling = hey_bob.rstrip().isupper()
    yell_question = question and yelling
    silent: re.Match[str] | None = re.match(r"^[\s\t\n]*$", hey_bob)

    if yell_question:
        return "Calm down, I know what I'm doing!"
    elif question:
        return "Sure."
    elif yelling:
        return "Whoa, chill out!"
    elif silent is not None:
        return "Fine. Be that way!"

    return "Whatever."

File: go/bob/bob.go
package bob

import (
	"regexp"
	"strings"
)

func Hey(remark string) string {
	remark = strings.TrimSpace(remark)
	question := regexp.MustCompile(`^.*\?$`).MatchString(remark)
	yelling := regexp.MustCompile(`^[^a-z]*[A-Z][^a-z]*!?$`).MatchString(remark)
	yellQuestion := regexp.MustCompile(`^[^a-z]*[A-Z][^a-z]*\?$`).MatchString(remark)
	silent := regexp.MustCompile(`\s+`).ReplaceAllString(remark, "")

	if silent == "" {
		return "Fine. Be that way!"
	} else if yellQuestion {
		return "Calm down, I know what I'm doing!"
	} else if yelling {
		return "Whoa, chill out!"
	} else if question {
		return "Sure."
	}

	return "Whatever."
}
